system_tests/fast_confirm_test.go

```diff
@@ -40,9 +44,109 @@ import (
+	defer cancelCtx()
+	builder, stakerA, cleanupBuilder, cleanupBackgroundTx := setupFastConfirmation(ctx, t)
+	defer cleanupBuilder()
+	defer cleanupBackgroundTx()
+
+	// Withdraw ETH from L2 to L1
+	gethhook.RequireHookedGeth()
+	arbSys, err := precompilesgen.NewArbSys(types.ArbSysAddress, builder.L2.Client)
+	Require(t, err)
+	authL2 := builder.L2Info.GetDefaultTransactOpts("User", ctx)
+	authL2.Value = big.NewInt(1000)
+	tx, err := arbSys.WithdrawEth(&authL2, common.Address{})
+	Require(t, err, "ArbSys failed")
+
+	// Wait for transaction to be executed
+	time.Sleep(4 * time.Millisecond) // Geth takes a few ms for the receipt to show up
+	var receipt *types.Receipt
+	receipt, err = builder.L2.Client.TransactionReceipt(ctx, tx.Hash())
+	Require(t, err, "No receipt for txn")
+	if receipt.Status != types.ReceiptStatusSuccessful {
+		Fatal(t, "Tx failed with status code:", receipt)
+	}
+	if len(receipt.Logs) == 0 {
+		Fatal(t, "Tx didn't emit any logs")
+	}
+
+	// Wait for staker to confirm the withdrawal
+	time.Sleep(time.Second)
+	tx, err = stakerA.Act(ctx)
+	Require(t, err)
+	if tx != nil {
+		_, err = builder.L1.EnsureTxSucceeded(tx)
+		Require(t, err)
+	}
+
+	arbSysAbi, err := precompilesgen.ArbSysMetaData.GetAbi()
+	Require(t, err, "failed to get abi")
+	withdrawTopic := arbSysAbi.Events["L2ToL1Tx"].ID
+	authL1 := builder.L1Info.GetDefaultTransactOpts("User", ctx)
+	nodeInterface, err := node_interfacegen.NewNodeInterface(types.NodeInterfaceAddress, builder.L2.Client)
+	Require(t, err)
+	merkleState, err := arbSys.SendMerkleTreeState(&bind.CallOpts{})
+	Require(t, err, "could not get merkle root")
+	bridgeBinding, err := bridgegen.NewBridge(builder.L1Info.GetAddress("Bridge"), builder.L1.Client)
+	Require(t, err)
+	outboxAddress, err := bridgeBinding.AllowedOutboxList(&bind.CallOpts{}, big.NewInt(0))
+	Require(t, err)
+	outboxBinding, err := bridgegen.NewOutbox(outboxAddress, builder.L1.Client)
+	Require(t, err)
+	// Check logs for withdraw event
+	for _, log := range receipt.Logs {
+		if log.Topics[0] == withdrawTopic {
+			parsedLog, err := arbSys.ParseL2ToL1Tx(*log)
+			Require(t, err, "Failed to parse log")
+
+			// Check NodeInterface.sol produces equivalent proofs
+			outboxProof, err := nodeInterface.ConstructOutboxProof(
+				&bind.CallOpts{}, merkleState.Size.Uint64(), parsedLog.Position.Uint64(),
+			)
+			Require(t, err)
+			// Execute the transaction on L1
+			execTx, err := outboxBinding.ExecuteTransaction(&authL1, outboxProof.Proof, parsedLog.Position, parsedLog.Caller, parsedLog.Destination, parsedLog.ArbBlockNum, parsedLog.EthBlockNum, parsedLog.Timestamp, parsedLog.Callvalue, parsedLog.Data)
+			Require(t, err)
+			time.Sleep(200 * time.Millisecond)
+			execReceipt, err := builder.L1.Client.TransactionReceipt(ctx, execTx.Hash())
+			Require(t, err, "No receipt for txn")
+			if execReceipt.Status != types.ReceiptStatusSuccessful {
+				Fatal(t, "Tx failed with status code:", receipt)
+			}
```

Comment:
Why not using EnsureTxSucceeded?
```suggestion
			execReceipt, err := builder.L1.EnsureTxSucceeded(execTx)
			Require(t, err)
```